fighters/snake/src/status.rs

```diff
@@ -0,0 +1,737 @@
+        MotionModule::change_motion(fighter.module_accessor, Hash40::new_raw(motion), 0.0, 1.0, false, 0.0, false, false);
+    }
+    else {
+        let motion = WorkModule::get_int64(fighter.module_accessor, *FIGHTER_SNAKE_STATUS_WORK_INT_MOT_AIR_KIND);
+        MotionModule::change_motion(fighter.module_accessor, Hash40::new_raw(motion), 0.0, 1.0, false, 0.0, false, false);
+    }
+    fighter.sub_shift_status_main(L2CValue::Ptr(special_lw_produce_main_loop as *const () as _))
+}
+pub unsafe fn special_lw_produce_main_loop(fighter: &mut L2CFighterCommon) -> L2CValue {
+    //println!("prev status is {}, {}", StatusModule::prev_status_kind(fighter.module_accessor, 0), StatusModule::prev_status_kind(fighter.module_accessor, 1));
+    if WorkModule::is_flag(fighter.module_accessor, *FIGHTER_SNAKE_STATUS_SPECIAL_LW_FLAG_SQUAT)
+    && StatusModule::situation_kind(fighter.module_accessor) == *SITUATION_KIND_AIR {
+        WorkModule::off_flag(fighter.module_accessor, *FIGHTER_SNAKE_STATUS_SPECIAL_LW_FLAG_SQUAT);
+        WorkModule::set_int64(fighter.module_accessor, hash40("special_lw_start") as i64, *FIGHTER_SNAKE_STATUS_WORK_INT_MOT_KIND);
+    }
+    if MotionModule::is_end(fighter.module_accessor) {
+        let entry_id = WorkModule::get_int(fighter.module_accessor, *FIGHTER_INSTANCE_WORK_ID_INT_ENTRY_ID) as usize;
+        if ControlModule::check_button_on(fighter.module_accessor, *CONTROL_PAD_BUTTON_CATCH)
+        || ControlModule::check_button_on(fighter.module_accessor, *CONTROL_PAD_BUTTON_GUARD) {
+            if StatusModule::situation_kind(fighter.module_accessor) == *SITUATION_KIND_AIR {
+                fighter.change_status(FIGHTER_SNAKE_STATUS_KIND_SPECIAL_LW_DROP.into(), false.into());
+            }
+            else {
+                fighter.change_status(FIGHTER_SNAKE_STATUS_KIND_SPECIAL_LW_ESTABLISH_GROUND.into(), false.into());
+            }
+            VarModule::on_flag(fighter.object(), vars::snake::instance::SELF_STICK);
+        }
+        else {
+            VarModule::off_flag(fighter.object(), vars::snake::instance::SELF_STICK);
+            fighter.change_status(FIGHTER_SNAKE_STATUS_KIND_SPECIAL_LW_ESTABLISH_JUDGE.into(), false.into());
+        }
+        false.into()
+    }
+    else {
+        fun_7100018800(fighter, false);
+        true.into()
+    }
+}
+pub unsafe fn fun_7100018800(fighter: &mut L2CFighterCommon, skip_check: bool) {
+    if skip_check == false {
+        if StatusModule::situation_kind(fighter.module_accessor) == StatusModule::prev_situation_kind(fighter.module_accessor) {
+            return
+        }
+    }
+    if StatusModule::situation_kind(fighter.module_accessor) == *SITUATION_KIND_GROUND {
+        GroundModule::set_rhombus_offset(fighter.module_accessor, &Vector2f{x:0.0, y:0.0});
+        KineticModule::change_kinetic(fighter.module_accessor, *FIGHTER_KINETIC_TYPE_GROUND_STOP);
+        fighter.set_situation(SITUATION_KIND_GROUND.into());
```

Comment:
*SITUATION_KIND_GROUND instead of .into()

Reply:
this caused compile errors, and it just suggested i do the old thing
![image](https://github.com/HDR-Development/HewDraw-Remix/assets/66037428/51d203e3-77c4-4dbb-a6a9-31ec05988643)